src/Maestro/SubscriptionActorService/SubscriptionActor.cs

```diff
@@ -110,9 +110,16 @@ public async Task UpdateForMergedPullRequestAsync(int updateBuildId)
         {
             Logger.LogInformation($"Updating {SubscriptionId} with latest build id {updateBuildId}");
             Subscription subscription = await Context.Subscriptions.FindAsync(SubscriptionId);
-            subscription.LastAppliedBuildId = updateBuildId;
-            Context.Subscriptions.Update(subscription);
-            await Context.SaveChangesAsync();
+            if (subscription != null)
```

Comment:
I guess there is no need to un-register as we do [here](https://github.com/dotnet/arcade-services/blob/master/src/Maestro/SubscriptionActorService/PullRequestActor.cs#L1107)?

Reply:
I don't think so, as we were not unregistering on the success case. This will simply be a no-op call, as there is nothing right now that we need to update for a deleted subscription.